arrow-cast/src/parse.rs

```diff
@@ -614,6 +629,19 @@ mod tests {
             naive_datetime.timestamp_nanos(),
             parse_timestamp("2020-09-08 13:42:29").unwrap()
         );
+
+        let tz: Tz = "+02:00".parse().unwrap();
+        let date = string_to_datetime(&tz, "2020-09-08 13:42:29").unwrap();
+        let utc = date.naive_utc().to_string();
+        assert_eq!(utc, "2020-09-08 11:42:29");
+        let local = date.naive_local().to_string();
+        assert_eq!(local, "2020-09-08 13:42:29");
+
+        let date = string_to_datetime(&tz, "2020-09-08 13:42:29Z").unwrap();
+        let utc = date.naive_utc().to_string();
+        assert_eq!(utc, "2020-09-08 13:42:29");
+        let local = date.naive_local().to_string();
+        assert_eq!(local, "2020-09-08 15:42:29");
```

Comment:
Add some tests to check timestamp is correct?
```suggestion
        assert_eq!(local, "2020-09-08 15:42:29");
        
        let &tz = Local::now().offset();
        let date = string_to_datetime(&tz, "2020-09-08T13:42:29Z").unwrap();
        let dt = NaiveDateTime::parse_from_str("2020-09-08T13:42:29Z","%Y-%m-%dT%H:%M:%SZ").unwrap();
        let ts = date.timestamp();
        assert_eq!(dt.timestamp(), ts);
        let date = string_to_datetime(&tz, "2020-09-08 13:42:29").unwrap();
        assert_eq!(dt.timestamp() - (tz.local_minus_utc() as i64), date.timestamp());
```

Reply:
I wanted to avoid using `Local::now` in the tests as this makes the tests not reproducible. The tests above should cover this, no?

Reply:
I guess  the tests above can run ok across machines, you can try.

Reply:
They do, but they don't consistently test the behaviour. For example my machine (and CI) is set to UTC, which means it won't test the behaviour at all...

Reply:
Maybe we can use two different time zone to test the behavior. For example:
```
 let tz1: Tz = "+02:00".parse().unwrap();
let tz2: Tz = "+08:00".parse().unwrap();
```

Reply:
That is OK, thx.